app/controllers/api/v1/sessions_controller.rb

```diff
@@ -8,6 +9,15 @@ class SessionsController < Devise::SessionsController
       # POST /resource/sign_in
       def create
         if params[:type] == 'facebook'
+          user = obtain_facebook_user(params[:fb_access_token])
+          render json: { error: 'Not Authorized' }, status: :forbidden and return if user.nil?
```

Comment:
Isn't better to raise an exception on the `obtain_facebook_user` method and that exception to be cached by the `render_forbidden_access` on the `ApiController`?